Code/GraphMol/RGroupDecomposition/RGroupCore.h

```diff
@@ -0,0 +1,41 @@
+//
+//  Copyright (C) 2020 Novartis Institutes for BioMedical Research
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#ifndef RGROUP_CORE
+#define RGROUP_CORE
+
+#include "../RDKitBase.h"
+#include "RGroupUtils.h"
+
+namespace RDKit {
+
+//! RCore is the core common to a series of molecules
+struct RCore {
+  boost::shared_ptr<RWMol> core;
+  boost::shared_ptr<RWMol> labelledCore;
+  std::set<int> core_atoms_with_user_labels;
```

Comment:
Have you considered using the more performant `std::unordered_set` here? It's basically my default unless I know I want ordered iteration, or if I can guarantee that the set will always be very small.

Reply:
This `std::set` is always going to be very small, so I don't think it would be make a big difference.

Reply:
@d-b-w Thanks for your comments!

Reply:
btw - from experience, the major speed cost is in the permutations and the scoring function.  I doubt you'll see most other optimizations.